### PR TITLE
coti-ethers

### PR DESCRIPTION
**Pull request by eitanz-coti**

### CODE REVIEW

File: src/crypto_utils.ts [hunk 1]
```diff
@@ -203,7 +199,7 @@ export function decryptString(ciphertext: Array<bigint>, userKey: string): strin
     return decoder.decode(new Uint8Array(decryptedStr))
 }
 
-export function generateAesKey(): string {
```
Comment (smiller-coti) [inline src/crypto_utils.ts:206]: I think we should just delete this function. It is not useful for anything other than one small part of the TS SDK examples. We can come up with a workaround

**Review by smiller-coti — CHANGES_REQUESTED** [review]
See comments

**Review by gmesika-coti — APPROVED** [review]

File: src/account/onboard.ts [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-
-function getDefaultContract(wallet: Signer) {
-    return new Contract(ONBOARD_CONTRACT_ADDRESS, JSON.stringify(ONBOARD_CONTRACT_ABI), wallet)
-}
-
-export async function onboard(user: BaseWallet, contract = getDefaultContract(user)) {
-    const {publicKey, privateKey} = generateRSAKeyPair()
-
-  const signedEK = sign(keccak256(publicKey), user.privateKey)
-  const receipt = await (await contract.onboardAccount(publicKey, signedEK, { gasLimit: 12000000 })).wait()
```
Comment (gmesika-coti) [inline src/account/onboard.ts:13]: gas limit should be a var

File: src/account/onboard.ts [hunk 3]
```diff
@@ -1,24 +0,0 @@
...
-  if (!receipt || !receipt.logs || !receipt.logs[0]) {
```
Comment (gmesika-coti) [inline src/account/onboard.ts:14]: we should check if receipt.status is 1 or not

File: src/account/onboard.ts [hunk 4]
```diff
@@ -1,24 +0,0 @@
-import {BaseWallet, Contract, keccak256, Signer} from "ethers"
-import {decryptRSA, generateRSAKeyPair, sign} from "../crypto_utils"
-import {ONBOARD_CONTRACT_ABI, ONBOARD_CONTRACT_ADDRESS} from "./onboard-contract"
-
-function getDefaultContract(wallet: Signer) {
-    return new Contract(ONBOARD_CONTRACT_ADDRESS, JSON.stringify(ONBOARD_CONTRACT_ABI), wallet)
-}
-
-export async function onboard(user: BaseWallet, contract = getDefaultContract(user)) {
-    const {publicKey, privateKey} = generateRSAKeyPair()
```
Comment (gmesika-coti) [inline src/account/onboard.ts:10]: plz mention in var name it is the rsa

**Review by gmesika-coti — CHANGES_REQUESTED** [review]

**Comment by gmesika-coti** [comment]
version #0.6.5

**Comment by gmesika-coti** [comment]
version #0.6.5

**Pull request closed without merging**